### PR TITLE
Added getting started guide for react native

### PR DESCRIPTION
**Pull request by timMikkelsen**

### CODE REVIEW

**Review by casm-mp — CHANGES_REQUESTED** [review]
docsite
* folder name space -> %20, use hyphen-seperated-name instead?
* should the files be numbered? It is ignored in the url, but easier to get an overview.

In general:
* maybe mention *where* to insert the code, so that we will get the same order when comparing with the complete version of the getting started project.
* Make sure the text describes why and what to change.
* App.tsx - remove comment about order of screens

end result:
* direction steps not shown on Android
* not properly switching between instruction steps

File: docs/01-getting-started/React Native/getting-directions.mdx [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+We will now add a `NavigationHeader` to the bottom sheet. The navigation header will be used to show the locations selected for getting directions. It's a simple view of two text fields that includes the selected locations name and venue as well as a button to get a route.
+
+```tsx
+const BoldText = (props: React.PropsWithChildren) => <Text style={{fontWeight: 'bold'}}>{props.children}</Text>
+
+type NavigationHeaderProps = {
+  searchResults: MPLocation[] | undefined;
+  fromLocation: MPLocation | undefined;
+  toLocation: MPLocation | undefined;
+  getRoute: () => void;
```
Comment (casm-mp) [inline docs/01-getting-started/React Native/getting-directions.mdx:77]: from/toLocation and getRoute should probably be present in the basic code

File: docs/01-getting-started/React Native/livedata.mdx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+* Changing the appearance of meeting rooms or workspace desks on a map, or in a list, based on occupancy information. For example, change the icon in order to indicate that a room is occupied.
+* Changing the position of a POI representing a vehicle.
+
+Support for Live Data requires that server-side integrations are in place. For example, visualizing live occupancy data requires that a calendar or booking system integration is in place. An integration like that is set up in [collaboration with MapsPeople](https://www.mapspeople.com/mapsindoors-integrations/).
+
+The following section relies on the existence of Live Position Data. If you do not have access to a MapsIndoors Dataset that has a Live Data integration, you should use our demo API key: `d876ff0e60bb430b8fabb145`.
+
+<br/>
+Enabling Live Data through [`MapControl`](https://app.mapsindoors.com/mapsindoors/reference/react-native/google-maps/1.0.0/classes/MapControl.html) is as simple as calling [`mapControl.enableLiveData()`](https://app.mapsindoors.com/mapsindoors/reference/react-native/google-maps/1.0.0/classes/MapControl.html#enableLiveData).
```
Comment (casm-mp) [inline docs/01-getting-started/React Native/livedata.mdx:23]: These links show the markdown syntax instead.

File: docs/01-getting-started/React Native/livedata.mdx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+```
+
+By consequence, [`MapControl`](https://app.mapsindoors.com/mapsindoors/reference/react-native/google-maps/1.0.0/classes/MapControl.html) is as simple as calling [`mapControl.enableLiveData()`](https://app.mapsindoors.com/mapsindoors/reference/react-native/google-maps/1.0.0/classes/MapControl.html) will manage the Live Data subscriptions needed for the currently visible map and provide a default rendering of the Live Data updates depending on the Domain Type.
+
+Using the demo API key you should now be able to see a "Staff Person" moving from one end to the other at ground floor in The White House main building.
+
+(TODO: new gif)<br/>
+Expected result:
+
+<img src="/img/getting-started/android_live_data_gif.gif" alt="An animation showing the desired behaviour of this tutorial" width="350"/>
```
Comment (casm-mp) [inline docs/01-getting-started/React Native/livedata.mdx:55]: We should update the video to be of the react native implementation

File: docs/01-getting-started/React Native/getting-directions.mdx
Comment (casm-mp) [inline docs/01-getting-started/React Native/getting-directions.mdx:1]: Some code is introduced, then changed further down:
	* Bottomsheet gets onClose added later.
	* Changed so it shows either instructions or results depending on which is available
	— The UI part is probably not as important, and we should focus on describing how we get the directions. So maybe we can just provide the final version to begin with.

File: docs/01-getting-started/React Native/project-setup.mdx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+### Folder structure
+
+To better seperate the typescript code from the other configuration related files, we move `App.tsx` into a new folder called `src/`. In order for the App to recognize this, we make the following change to `index.js`:
+
+```diff
+- import App from './App';
++ import App from './src/App';
+```
```
[the remark this answers is not in the document]
Comment (timMikkelsen) [reply docs/01-getting-started/React Native/project-setup.mdx]: Made a clearer seperation of the basic/manual setup.
npm install/pod install is described in the readme of the code projects. As mentioned in the step for the basic example.
Pager view is not used in the project.

File: docs/01-getting-started/React Native/search.mdx [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+        <SearchBox onSearch={search}/>
+      </View>
+      <BottomSheet ref={bottomSheet} snapPoints={['15%', '60%']} index={-1} enablePanDownToClose={true} onChange={() => {}} onClose={() => {setSearchResults(undefined)}}>
+        <SearchResults searchResults={searchResults} clickResult={clickResult}/>
+      </BottomSheet>
+    </GestureHandlerRootView>
+  );
+```
+
+The components have already been made, but we need to fill in some data to show our search results from the `MPLocation`. The SearchBox is just a simple input field with two buttons, to call our search function with a string value from the input box.
```
[the remark this answers is not in the document]
Comment (timMikkelsen) [reply docs/01-getting-started/React Native/search.mdx]: Added more description to these steps now

File: docs/01-getting-started/React Native/search.mdx [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+              height: 48,
+              backgroundColor: Colors.white,
+              borderColor: Colors.dark,
+              borderWidth: 1,
+              paddingHorizontal: 5,
+              flex: 1,
+            }}
+        />
+        <Button title={'Search'} onPress={() => onSearch(searchBoxText)}/>
+        <Button title={'Cancel'} onPress={() => onSearch(undefined)}/>
```
[the remark this answers is not in the document]
Comment (timMikkelsen) [reply docs/01-getting-started/React Native/search.mdx:106]: It is now present in the basic example code

File: docs/01-getting-started/React Native/search.mdx
[the remark this answers is not in the document]
Comment (timMikkelsen) [reply docs/01-getting-started/React Native/search.mdx:1]: Added more detail to where to insert cod, in gerneral

File: docs/01-getting-started/React Native/display-a-map.mdx
[the remark this answers is not in the document]
Comment (timMikkelsen) [reply docs/01-getting-started/React Native/display-a-map.mdx:1]: More descriptive now

File: docs/01-getting-started/React Native/display-a-map.mdx [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+  const {width, height} = useWindowDimensions();
+
+  return (
+    <GestureHandlerRootView style={{flex:1, flexGrow:1}}>
+      <MapView style={{
+        width: width,
+        height: height,
+        flex: 1,
+      }}/>
```
[the remark this answers is not in the document]
Comment (timMikkelsen) [reply docs/01-getting-started/React Native/display-a-map.mdx:135]: Empty bottom sheet is now a part of the code example

File: docs/01-getting-started/React Native/getting-directions.mdx (hunk 1 above)
Comment (timMikkelsen) [reply docs/01-getting-started/React Native/getting-directions.mdx:81]: It is specifically explained, so no need.

File: docs/01-getting-started/React Native/getting-directions.mdx
Comment (timMikkelsen) [reply docs/01-getting-started/React Native/getting-directions.mdx:1]: Code changes are explained specifically now

**Comment by timMikkelsen** [comment]
> docsite
> 
> * folder name space -> %20, use hyphen-seperated-name instead?
> * should the files be numbered? It is ignored in the url, but easier to get an overview.
> 
> In general:
> 
> * maybe mention _where_ to insert the code, so that we will get the same order when comparing with the complete version of the getting started project.
> * Make sure the text describes why and what to change.
> * App.tsx - remove comment about order of screens
> 
> end result:
> 
> * direction steps not shown on Android
> * not properly switching between instruction steps

Keeping the same naming as for every other folder.
We'll keep them as is for now

More specifically mentioned where to insert code, will not mention specific order/lines as it is not necessary in my opinion, and files are small enough that a manual comparison should be sufficient. 
Done

end results part was addressed in office...

**Review by casm-mp — CHANGES_REQUESTED** [review]
Smaller comments.

Also maybe change the folder name to 'react-native' to match the other guides, and avoids a %20 in the url.

**Review by casm-mp — APPROVED** [review]

**Review by matiasholst — APPROVED** [review]

**Pull request merged**